pkg/providers/v1/aws.go

```diff
@@ -1503,6 +1503,23 @@ func (c *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.No
 			}
 		}
 
+		for _, macID := range macIDs {
+			ipv6Path := path.Join("network/interfaces/macs/", macID, "ipv6s")
+			internalIPv6s, err := c.metadata.GetMetadata(ipv6Path)
+			if err != nil {
+				// Interfaces without an IPv6 address don't get the "ipv6s" data
+				klog.V(4).Infof("error querying AWS metadata for %q: %q", ipv6Path, err)
+				continue
+			}
+
+			for _, internalIPv6 := range strings.Split(internalIPv6s, "\n") {
+				if internalIPv6 == "" {
+					continue
+				}
+				addresses = append(addresses, v1.NodeAddress{Type: v1.NodeInternalIP, Address: internalIPv6})
```

Comment:
> Should we should use it only if ip.IsGlobalUnicast()

Fwiw, I think we should use whatever address(es?) are reported by VPC (in metadata).  If VPC is giving us ULA addresses (for example), then there's a very good reason, and the code here should trust the VPC address is relevant and use it.

I have no reason to believe this will ever happen to be clear! but if VPC _is_ reporting something 'weird', then I think there's a very good chance that VPC is correct for that situation and our `if IsGlobalUnicast()` logic is going to be incorrect.

Reply:
ULA are `IsGlobalUnicast()` , this just removes link-local, multicast and loopback
But agreed, this validation shouldn't be here and VPC is the source of truth